lib/packwerk/parse_run.rb

```diff
@@ -76,5 +84,20 @@ def run(reference_lister)
       @progress_formatter.finished(execution_time)
       all_offenses
     end
+
+    def update_progress(failed: false)
+      if failed
+        @progress_formatter.mark_as_failed
+      else
+        @progress_formatter.mark_as_inspected
+      end
+    end
+
+    def includes_unlisted_offense?(offenses, reference_lister)
+      offenses.any? do |offense|
+        next true unless offense.is_a?(ReferenceOffense)
```

Comment:
This same block is now duplicated inside the same called `filtered_offenses`. Should not it be possible to execute this block inside `find_offenses` so we don't need to call it twice?

Reply:
Want me to remove the second filtering stage for now and reintroduce it in the next PR when parallel is brought in?